samples/bluetooth/peripheral_power_profiling/src/main.c

```diff
@@ -0,0 +1,748 @@
+}
+
+static const struct pwr_service_cb pwr_service_callback = {
+	.notify_complete_cb = pwr_notification_cb,
+	.evt_cb = pwr_event_cb,
+};
+
+static void pairing_failed(struct bt_conn *conn, enum bt_security_err reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing failed conn: %s, reason %d\n", addr, reason);
+
+	/* Generate new pairing keys if pairing procedure failed. */
+	k_work_submit(&key_generate_work);
+}
+
+static struct bt_conn_auth_info_cb conn_auth_info_callbacks = {
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed,
+};
+
+static void button_handler(uint32_t button_state, uint32_t has_changed)
+{
+	int err;
+	uint32_t buttons = button_state & has_changed;
+
+	if (buttons & CONNECTABLE_ADV_BUTTON) {
+		struct bt_le_ext_adv_start_param connectable_start_param = {
+			.timeout = CONNECTABLE_ADV_TIMEOUT,
+			.num_events = 0
+		};
+
+		/* It is called from workqueue context, checking return value is not needed. */
+		k_work_cancel_delayable(&system_off_work);
+
+		(void)bt_le_ext_adv_stop(adv_set);
+		err = bt_le_ext_adv_update_param(adv_set, connectable_ad_params);
+		if (err) {
+			printk("Failed to set connectable advertising data (err %d)\n", err);
+			return;
+		}
+
+		err = bt_le_ext_adv_set_data(adv_set, connectable_ad_data,
+				     ARRAY_SIZE(connectable_ad_data), NULL, 0);
+		if (err) {
+			printk("Failed to set data for connectable advertising (err %d)\n", err);
+			return;
+		}
+
+		err = bt_le_ext_adv_start(adv_set, &connectable_start_param);
+		if (err) {
+			printk("Connectable advertising failed to start (err %d)\n", err);
+		} else {
+			printk("Connectable advertising started\n");
+		}
+
+		return;
+	}
+
+	if (buttons & NON_CONNECTABLE_ADV_BUTTON) {
+		struct bt_le_ext_adv_start_param non_connectable_start_param = {
+			.timeout = NON_CONNECTABLE_ADV_TIMEOUT,
+			.num_events = 0
+		};
+
+		/* It is called from workqueue context, checking return value is not needed. */
+		k_work_cancel_delayable(&system_off_work);
+
+		(void)bt_le_ext_adv_stop(adv_set);
+		err = bt_le_ext_adv_update_param(adv_set, non_connectable_ad_params);
+		if (err) {
+			printk("Failed to set non-connectable advertising data (err %d)\n", err);
+			return;
+		}
+
+		err = bt_le_ext_adv_set_data(adv_set, non_connectable_ad_data,
+					     ARRAY_SIZE(non_connectable_ad_data), NULL, 0);
+		if (err) {
+			printk("Failed to set data for non-connectable advertising (err %d)\n",
+			       err);
+			return;
+		}
+
+		err = bt_le_ext_adv_start(adv_set, &non_connectable_start_param);
+		if (err) {
+			printk("Non-connectable advertising failed to start (err %d)\n", err);
+		} else {
+			printk("Non-connectable advertising started\n");
+		}
+
+		return;
+	}
+}
+
+static int tk_value_generate(void)
+{
+	int err;
+
+	err = bt_rand(tk_local, sizeof(tk_local));
+	if (err) {
+		printk("Failed to generate random TK value (err %d)\n", err);
+	}
+
+	return err;
+}
```

Comment:
Considering removing this function and moving its logic to the pairing_key_generate function since its only called from there

Reply:
Done